src/meta/src/rpc/ddl_controller.rs

```diff
@@ -111,13 +113,81 @@ pub struct DdlController<S: MetaStore> {
     barrier_manager: BarrierManagerRef<S>,
 
     aws_client: Arc<Option<AwsEc2Client>>,
+    // The semaphore is used to limit the number of concurrent streaming job creation.
+    creating_streaming_job_permits: Arc<CreatingStreamingJobPermit<S>>,
+}
+
+#[derive(Clone)]
+pub struct CreatingStreamingJobPermit<S: MetaStore> {
+    semaphore: Arc<Semaphore>,
+    _phantom: std::marker::PhantomData<S>,
+}
+
+impl<S> CreatingStreamingJobPermit<S>
+where
+    S: MetaStore,
+{
+    async fn new(env: &MetaSrvEnv<S>) -> Self {
+        let mut permits = env
+            .system_params_manager()
+            .get_params()
+            .await
+            .max_concurrent_creating_streaming_jobs() as usize;
+        if permits == 0 {
+            // if the system parameter is set to zero, use the max permitted value.
+            permits = Semaphore::MAX_PERMITS;
+        }
+        let semaphore = Arc::new(Semaphore::new(permits));
+
+        let (local_notification_tx, mut local_notification_rx) =
+            tokio::sync::mpsc::unbounded_channel();
+        env.notification_manager()
+            .insert_local_sender(local_notification_tx)
+            .await;
+        let semaphore_clone = semaphore.clone();
+        tokio::spawn(async move {
+            while let Some(notification) = local_notification_rx.recv().await {
+                let LocalNotification::SystemParamsChange(p) = &notification else {
+                    continue;
+                };
+                let mut new_permits = p.max_concurrent_creating_streaming_jobs() as usize;
+                if new_permits == 0 {
+                    new_permits = Semaphore::MAX_PERMITS;
+                }
+                match permits.cmp(&new_permits) {
+                    Ordering::Less => {
+                        semaphore_clone.add_permits(new_permits - permits);
+                    }
+                    Ordering::Equal => continue,
+                    Ordering::Greater => {
+                        semaphore_clone
+                            .acquire_many((permits - new_permits) as u32)
```

Comment:
`acquire_many` here might take a while for it to take effect if there are some running creating jobs. But I think it's acceptable.